Python/Movie-Info-Telegram-Bot/bot.py

```diff
@@ -0,0 +1,153 @@
+        +'\nSend /name movie_name to know the genre and rating of the movie.\nSend /genre genre_name to'
+        +'get the list of movies belonging to that genre'
+    )
+
+def help(update, context):
+    """Send a message when the command /help is issued."""
+    update.message.reply_text('Help!')
+
+def genre(update, context):
+    """Send a list of movies when the command /genre is issued."""
+    url = 'https://www.imdb.com/search/title/'
+    genre= str(update.message.text)[7:]
+    print(genre)
+    r=requests.get(url+'?genres='+genre)
+    soup=BeautifulSoup(r.text,"html.parser")
+    title = soup.find('title')
+    if title.string == 'IMDb: Advanced Title Search - IMDb':
+        update.message.reply_text("Sorry,No such genre.Try again")
+    else:
+        res=[]
+        res.append(title.string+'\n')
+        tags = soup('a')
+        for tag in tags:
+            movie=re.search('<a href=\"/title/.*>(.*?)</a>',str(tag))
+            try:
+                if "&amp;" in movie.group(1):
+                    movie.group(1).replace("&amp;","&")
+                res.append(movie.group(1))
+            except :
+                pass
+        stri=""
+        for i in res:
+            stri+=i+'\n'
+        update.message.reply_text(stri)
+
+def name(update, context):
+    """Send the first 3 search results of the movie name in IMDb site when the command /name is issued."""
+    movie = str(update.message.text)[6:]
+    print(movie)
+    res=get_info(movie)
+    stri=""
+    for i in res:
+        for a in i:
+            stri+=a+'\n'
+        stri+='\n'
+    update.message.reply_text(stri)
+
+def error(update, context):
+    """Log Errors caused by Updates."""
+    logger.warning('Update "%s" caused error "%s"', update, context.error)
+
+def get_info(movie):
+    "To scrape IMDb and get genre and rating of the movie "
+    url = 'https://www.imdb.com/find?q='
+    r=requests.get(url+movie+'&ref_=nv_sr_sm')
+    soup=BeautifulSoup(r.text,"html.parser")
+    title = soup.find('title')
+    tags = soup('a')
+    pre_url=""
+    count=0
+    lis=[]
+    res=[]
+    for tag in tags:
+        if(count>2):
+            break
+        m=re.search('<a href=.*>(.*?)</a>',str(tag))
+        try:
+            lis=[]
+            link=re.search('/title/(.*?)/',str(m))
+            new_url='https://www.imdb.com'+str(link.group(0))
+            if new_url!=pre_url:
+                html=requests.get(new_url)
+                soup_each_title=BeautifulSoup(html.text,"html.parser")
+                movie_title=soup_each_title.find('title').string.replace('- IMDb',' ')
+                anchor=soup_each_title('a')
+                genre_string="Genre : "
+                for each in anchor:
+                    genre=re.search('<a href=\"/search/title\?genres=.*> (.*?)</a>',str(each))
+                    try:
+                       genre_string+=genre.group(1)+' '
+                    except:
+                        pass
```

Comment:
@aish2002 The indentation for the `pass` statement doesn't seem right to me.
Or maybe its the `genre_string+=genre.group(1)+' '` statement (most probably)